src/odemis/acq/stream/_helper.py

```diff
@@ -1080,6 +1071,216 @@ def _onPolarizationMove(self, f):
+        wllist = self.getPixelToWavelength(data)
+        md[model.MD_WL_LIST] = wllist
+
+        if self.wl_inverted:
+            data = data[:,::-1, ...]  # invert C
+
+        # Sets POS and PIXEL_SIZE from the e-beam (which is in spot mode). Useful when taking snapshots.
+        epxs = self.emitter.pixelSize.value
+        md[model.MD_PIXEL_SIZE] = epxs
+        emd = self.emitter.getMetadata()
+        pos = emd.get(model.MD_POS, (0, 0))
+        trans = self.emitter.translation.value
+        md[model.MD_POS] = (pos[0] + trans[0] * epxs[0],
+                            pos[1] - trans[1] * epxs[1])  # Y is inverted
+
+        data = model.DataArray(data, metadata=md)
+        super(AngularSpectrumSettingsStream, self)._onNewData(dataflow, data)
+
+    def _find_metadata(self, md):
+        """
+        Finds the useful metadata for a 2D spatial projection from the metadata of a raw image.
+        :returns: (dict) Metadata dictionary (MD_* -> value).
+        """
+        simple_md = super(AngularSpectrumSettingsStream, self)._find_metadata(md)
+        if model.MD_THETA_LIST in md:
+            simple_md[model.MD_THETA_LIST] = md[model.MD_THETA_LIST]
+        if model.MD_WL_LIST in md:
+            simple_md[model.MD_WL_LIST] = md[model.MD_WL_LIST]
+        return simple_md
+
+    def _is_active_setter(self, active):
+        """
+        Called when stream is active/playing. Links the angular and spectrum binning VAs to
+        the camera resolution VA and the detector binning depending on whether the stream
+        is active or not.
+        :param active: (boolean) True if stream is playing.
+        :returns: (boolean) If stream is playing or not.
+        """
+        self._active = super(AngularSpectrumSettingsStream, self)._is_active_setter(active)
+
+        if self._active:
+            self._linkBin2CamRes()
+        else:
+            self._unlinkBin2CamRes()
+        return self._active
+
+    def _linkBin2CamRes(self):
+        """
+        Subscribes the detector resolution and binning to the spectrum and angular binning VAs.
+        """
+        self.angular_binning.subscribe(self._onBinning, init=True)
+        self.spectrum_binning.subscribe(self._onBinning, init=True)
+
+    def _unlinkBin2CamRes(self):
+        """
+        Unsubscribes the detector resolution and binning and update the GUI
+        """
+        self.angular_binning.unsubscribe(self._onBinning)
+        self.spectrum_binning.unsubscribe(self._onBinning)
+
+    def _onBinning(self, _=None):
+        """
+        Callback, which updates the binning on the detector and calculates spectral resolution
+        based on the spectrum and angular binning values.
+        Only called when stream is active.
+        """
+        try:
+            self._detector.binning.value = (self.spectrum_binning.value, self.angular_binning.value)
+        except Exception:
```

Comment:
As in the previous comment of Thera
https://github.com/delmic/odemis/pull/1921/commits/117818133ea2e79e297e903e773f5ab35c36b611#r880290106